src/Gui/Tree.cpp

```diff
@@ -351,34 +352,128 @@ class DocumentItem::ExpandInfo :
+    // rendering as transparent overlay.
+    bool trimBG = TreeParams::getHideColumn();
+    QRect rect = opt.rect;
+
+    if (index.column() == 0) {
+        if (tree->testAttribute(Qt::WA_NoSystemBackground)
+                && (trimBG || (opt.backgroundBrush.style() == Qt::NoBrush
+                                && _TreeItemBackground.style() != Qt::NoBrush)))
+        {
+            const int margin = style->pixelMetric(QStyle::PM_FocusFrameHMargin, &option, tree) + 1;
+            // 2 margin for text, 2 margin for decoration (icon)
+            int width = 4*margin + opt.fontMetrics.boundingRect(opt.text).width()
+                + opt.decorationSize.width() + TreeParams::getItemBackgroundPadding();
+            if (TreeParams::getCheckBoxesSelection()) {
+                // another 2 margin for checkbox
+                width += 2*margin + style->pixelMetric(QStyle::PM_IndicatorWidth)
+                    + style->pixelMetric(QStyle::PM_LayoutHorizontalSpacing);
+            }
+            if (width < rect.width())
+                rect.setWidth(width);
+            if (trimBG) {
+                rect.setWidth(rect.width() + 5);
+                opt.rect = rect;
+                if (opt.backgroundBrush.style() == Qt::NoBrush)
+                    painter->fillRect(rect, _TreeItemBackground);
+            } else if (!opt.state.testFlag(QStyle::State_Selected))
+                painter->fillRect(rect, _TreeItemBackground);
+        }
+
+    }
+
+    style->drawControl(QStyle::CE_ItemViewItem, &opt, painter, tree);
+}
+
+void TreeWidgetItemDelegate::initStyleOption(QStyleOptionViewItem *option,
+                                             const QModelIndex &index) const
+{
+    inherited::initStyleOption(option, index);
+
+    TreeWidget * tree = static_cast<TreeWidget*>(parent());
+    QTreeWidgetItem * item = tree->itemFromIndex(index);
+    if (!item || item->type() != TreeWidget::ObjectType)
+        return;
+
+    QSize size;
+    size = option->icon.actualSize(QSize(0xffff, 0xffff));
+    if (size.height())
+        option->decorationSize = QSize(size.width()*TreeWidget::iconSize()/size.height(),
+                                       TreeWidget::iconSize());
+}
+
+QWidget* TreeWidgetItemDelegate::createEditor(
+        QWidget *parent, const QStyleOptionViewItem &, const QModelIndex &index) const
 {
     auto ti = static_cast<QTreeWidgetItem*>(index.internalPointer());
-    if (ti->type() != TreeWidget::ObjectType || index.column() > 1)
-        return nullptr;
-    auto item = static_cast<DocumentObjectItem*>(ti);
-    App::DocumentObject* obj = item->object()->getObject();
-    auto& prop = index.column() ? obj->Label2 : obj->Label;
+    if(ti->type()!=TreeWidget::ObjectType || index.column()>1)
+        return 0;
+    DocumentObjectItem *item = static_cast<DocumentObjectItem*>(ti);
+    App::DocumentObject *obj = item->object()->getObject();
+    auto &prop = index.column()?obj->Label2:obj->Label;

```

Comment:
Looks like mistakenly-included code that overwrites some formatting improvements.